geth/utils/validation.py

```diff
@@ -0,0 +1,118 @@
+from __future__ import (
+    annotations,
+)
+
+from typing import (
+    Any,
+    Literal,
+)
+
+from pydantic import (
+    BaseModel,
+    ConfigDict,
+)
+
+
+class GethKwargs(BaseModel):
```

Comment:
I wonder actually if we should use a pattern like 

```py
from pydantic import create_model

# create a type for geth kwargs based off TypedDict so we can use `Unpack[GethKwargs]` in method signatures
class GethKwargs(TypedDict):
    arg1: str
    arg2: bool
    # etc...

# create the pydantic model for validation    
GethKwargsModel = create_model('GethKwargsModel', **GethKwargs)
```

... and then we can use `Unpack[GethKwargs]` instead of `Any` for typing them in method signatures. Any thoughts on that or alternate ways to not have to use `Any`?

Reply:
Kind of pseudocode, but unpack the kwargs into a pydantic model separately

```py
kwargs_fields = {k: (v, ...) for k, v in GethKwargs.__annotations__.items()}

# dynamically create a pydantic model from the `TypedDict`
GethKwargsModel = create_model('GethKwargsModel', **kwargs_fields)
```

Reply:
Discussed elsewhere, but validating with a GethKwargs TypedDict causes argument overlap issues in functions such as `get_accounts` that have args that are also keys in `geth_kwargs`. Shelving for now.